pkg/gorilla/gorilla.go

```diff
@@ -206,15 +215,28 @@ func (ws *WebSocket) initialize() {
 					ws.logger.LogChannel <- err.Error()
 					continue
 				}
-				responseChan, ok := ws.getResponseChannel(fmt.Sprintf("%v", res.ID))
-				if !ok {
-					err = errors.New("ResponseChannel is not ok")
-					ws.logger.Logger.Err(err)
-					ws.logger.LogChannel <- err.Error()
-					continue
+				var responseChan chan rpc.RPCResponse
+				var ok bool
+				if res.ID != nil && res.ID != "" {
+					responseChan, ok = ws.getResponseChannel(fmt.Sprintf("%v", res.ID))
+					if !ok {
+						err = errors.New("ResponseChannel is not ok")
+						ws.logger.Logger.Err(err)
+						ws.logger.LogChannel <- err.Error()
+						continue
+					}
+					responseChan <- res
+					close(responseChan)
+				} else {
+					responseChan, ok = ws.getResponseChannel(fmt.Sprintf("%v", res.Result.(map[string]interface{})["id"]))
```

Comment:
```suggestion
          resolved_id, ok = res.Result.(map[string]interface{})["id"]
          if !ok {
              err := errors.New("response did not contain an 'id' field")
              log.WithField("result", res.Result).Errorf("Failed to handle response", err)
              return err
          }
					responseChan, ok = ws.getResponseChannel(fmt.Sprintf("%v", res.Result.(map[string]interface{})["id"]))
```

Reply:
I am a bit confused about `log.WithField()`. we are not using the `log` package here, did you mean something like `ws.logger.Logger`?

Reply:
Yeah, I think he meant `ws.logger.Logger`
also, snake case are not usually used in go for various mostly files and constants that why golang-linter giving error.

Reply:
Would the log package not work?

Reply:
its not working. cant see WithField method as well.